docs/reference/files/file-libname-py.md

```diff
@@ -0,0 +1,160 @@
+(file-libname-py)=
+# File `<libname>.py`
+
+File `<libname>.py` is a Python file (or a Python [module](https://realpython.com/lessons/scripts-modules-packages-and-libraries/)) in your charm project that holds a charm library -- that is, code that enables charm developers to easily share and reuse auxiliary logic related to  charms -- for example, logic related to the relations between charms. 
+
+Authors associate libraries with a specific charm, and publish them to Charmhub with a reference to the origin charm. This does not prevent reuse, modification, and sharing.
+
+The publishing tools around libraries are deliberately kept simple. Libraries are however versioned and uniquely identified.
+
+
+<!--
+SEE IF WE CAN INTEGRATE THIS CONTENT FROM THIS DOC: https://juju.is/docs/sdk/document-your-library 
+One of the most distinct features of a charm is the way it can readily relate to another charm. For this interface, there is one charm that will provide information in a specific way, and a charm that will consume information in a matching way to connect and work together. To ensure that the charm interface is a perfect pairing, the operator framework tooling and the store support a mechanism to publish and reuse this logic in a form of simple files, that we’ll call libraries. 
+-->
+
+
+## Location
+
+Charm libraries are located in a  directory inside the charm with the following structure:
```

Comment:
For safety, we might want to check these files for `\b {2,}\b`

Reply:
Corrected in the rst version